Microsoft.Research/Contracts/System.Windows.Forms/System.Windows.Forms.AutoCompleteStringCollection.cs

```diff
@@ -0,0 +1,236 @@
+        public void AddRange(string[] value)
+        {
+            Contract.Requires(value != null);
+            Contract.Ensures(this.Count == Contract.OldValue(this.Count) + value.Length);
+        }
+
+        /// <summary>
+        /// Removes all strings from the collection.
+        /// </summary>
+        public void Clear()
+        {
+            Contract.Ensures(this.Count == 0);
+        }
+
+        // <summary>
+        // Indicates whether the <see cref="T:System.String"/> exists within the collection.
+        // </summary>
+        // 
+        // <returns>
+        // true if the <see cref="T:System.String"/> exists within the collection; otherwise, false.
+        // </returns>
+        // <param name="value">The <see cref="T:System.String"/> for which to search.</param>
+        // public bool Contains(string value)
+       
+        /// <summary>
+        /// Copies an array of <see cref="T:System.String"/> objects into the collection, starting at the specified position.
+        /// </summary>
+        /// <param name="array">The <see cref="T:System.String"/> objects to add to the collection.</param><param name="index">The position within the collection at which to start the insertion. </param>
+        public void CopyTo(string[] array, int index)
+        {
+            Contract.Requires(array != null);
+            Contract.Requires(index >= 0 && index <= this.Count);
+        }
+
+        /// <summary>
+        /// Obtains the position of the specified string within the collection.
+        /// </summary>
+        /// 
+        /// <returns>
+        /// The index for the specified item.
+        /// </returns>
+        /// <param name="value">The <see cref="T:System.String"/> for which to search.</param>
+        public int IndexOf(string value)
+        {
+            Contract.Ensures(Contract.Result<int>() >= -1);
+            return default(int);
+        }
+
+        /// <summary>
+        /// Inserts the string into a specific index in the collection.
+        /// </summary>
+        /// <param name="index">The position at which to insert the string.</param><param name="value">The string to insert.</param>
+        public void Insert(int index, string value)
+        {
+            Contract.Requires(index >= 0 && index <= this.Count);
+        }
+
+        // <summary>
+        // Removes a string from the collection.
+        // </summary>
+        // <param name="value">The <see cref="T:System.String"/> to remove.</param>
+        // public void Remove(string value)
+        
+        // <summary>
+        // Removes the string at the specified index.
+        // </summary>
+        // <param name="index">The zero-based index of the string to remove.</param>
+        public void RemoveAt(int index)
+        {
+            Contract.Requires(index >= 0 && index < this.Count);
+            Contract.Ensures(this.Count == Contract.OldValue(this.Count)-1);
```

Comment:
spacing: shoudl be `Contract.OldValue(Count) - 1`.

and can we avoid using `this`?